### PR TITLE
Add DNX451 to EntityFramework.Core

### LINKED ISSUE #3553: Unable to move from build 16018 to 16105 on aspnetcidev
**Failed to add reference to 'System.Collections.Concurrent'. Please make sure that it is in the Global Assembly Cache.**

With the SQL CE provider project: .NET 4.5.1 Class Library in VS 2015 with NuGet extension version: 3.2.60914, I get the error, when only updating Microsoft.Extensions.Logging from aspnetcidev, blocking me from moving to the latest RC1 bits. Also installing from a "empty" project with no NuGet dependencies causes the same error. 

PM> Install-Package Microsoft.Extensions.Logging -Pre
Attempting to gather dependencies information for package 'Microsoft.Extensions.Logging.1.0.0-rc1-15741' with respect to project 'EntityFramework.SqlServerCompact35', targeting '.NETFramework,Version=v4.5.1'
Attempting to resolve dependencies for package 'Microsoft.Extensions.Logging.1.0.0-rc1-15741' with DependencyBehavior 'Lowest'
Resolving actions to install package 'Microsoft.Extensions.Logging.1.0.0-rc1-15741'
Resolved actions to install package 'Microsoft.Extensions.Logging.1.0.0-rc1-15741'
Removed package 'Microsoft.Extensions.DependencyInjection.Abstractions.1.0.0-rc1-15705' from 'packages.config'
Successfully uninstalled 'Microsoft.Extensions.DependencyInjection.Abstractions.1.0.0-rc1-15705' from EntityFramework.SqlServerCompact35
Removed package 'Microsoft.Extensions.Logging.1.0.0-rc1-15704' from 'packages.config'
Successfully uninstalled 'Microsoft.Extensions.Logging.1.0.0-rc1-15704' from EntityFramework.SqlServerCompact35
Removed package 'Microsoft.Extensions.Logging.Abstractions.1.0.0-rc1-15704' from 'packages.config'
Successfully uninstalled 'Microsoft.Extensions.Logging.Abstractions.1.0.0-rc1-15704' from EntityFramework.SqlServerCompact35
Adding package 'Microsoft.Extensions.DependencyInjection.Abstractions.1.0.0-rc1-15736' to folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Added package 'Microsoft.Extensions.DependencyInjection.Abstractions.1.0.0-rc1-15736' to folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Added package 'Microsoft.Extensions.DependencyInjection.Abstractions.1.0.0-rc1-15736' to 'packages.config'
Successfully installed 'Microsoft.Extensions.DependencyInjection.Abstractions 1.0.0-rc1-15736' to EntityFramework.SqlServerCompact35
Adding package 'Microsoft.Extensions.Logging.Abstractions.1.0.0-rc1-15741' to folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Added package 'Microsoft.Extensions.Logging.Abstractions.1.0.0-rc1-15741' to folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Added package 'Microsoft.Extensions.Logging.Abstractions.1.0.0-rc1-15741' to 'packages.config'
Successfully installed 'Microsoft.Extensions.Logging.Abstractions 1.0.0-rc1-15741' to EntityFramework.SqlServerCompact35
Adding package 'Microsoft.Extensions.Logging.1.0.0-rc1-15741' to folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Added package 'Microsoft.Extensions.Logging.1.0.0-rc1-15741' to folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Install failed. Rolling back...
Package 'Microsoft.Extensions.Logging 1.0.0-rc1-15741' does not exist in project 'EntityFramework.SqlServerCompact35'
Removed package 'Microsoft.Extensions.Logging.Abstractions 1.0.0-rc1-15741' from 'packages.config'
Removed package 'Microsoft.Extensions.DependencyInjection.Abstractions 1.0.0-rc1-15736' from 'packages.config'
Package 'Microsoft.Extensions.Logging.Abstractions.1.0.0-rc1-15704' already exists in folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Added package 'Microsoft.Extensions.Logging.Abstractions.1.0.0-rc1-15704' to 'packages.config'
Package 'Microsoft.Extensions.Logging.1.0.0-rc1-15704' already exists in folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Added package 'Microsoft.Extensions.Logging.1.0.0-rc1-15704' to 'packages.config'
Package 'Microsoft.Extensions.DependencyInjection.Abstractions.1.0.0-rc1-15705' already exists in folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Added package 'Microsoft.Extensions.DependencyInjection.Abstractions.1.0.0-rc1-15705' to 'packages.config'
Removing package 'Microsoft.Extensions.Logging 1.0.0-rc1-15741' from folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Removed package 'Microsoft.Extensions.Logging 1.0.0-rc1-15741' from folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Removing package 'Microsoft.Extensions.Logging.Abstractions 1.0.0-rc1-15741' from folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Removed package 'Microsoft.Extensions.Logging.Abstractions 1.0.0-rc1-15741' from folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Removing package 'Microsoft.Extensions.DependencyInjection.Abstractions 1.0.0-rc1-15736' from folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Removed package 'Microsoft.Extensions.DependencyInjection.Abstractions 1.0.0-rc1-15736' from folder 'C:\Code\EntityFramework7.SqlServerCompact\packages'
Install-Package : **Failed to add reference to 'System.Collections.Concurrent'. Please make sure that it is in the Global Assembly Cache.**
At line:1 char:1
- Install-Package Microsoft.Extensions.Logging -Pre
- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  - CategoryInfo          : NotSpecified: (:) [Install-Package], Exception
  - FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManagement.PowerShellCmdlets.InstallPackageCommand

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #3553

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @bricelam \ @natemcmaster  \ @davidfowl

**Comment by sebastienros** [comment]
:shipit:

**Comment by davidfowl** [comment]
:cry:

**Comment by pranavkm** [comment]
@bricelam \ @natemcmaster ?

File: src/EntityFramework.Core/project.json [hunk 1]
```diff
@@ -28,6 +28,25 @@
   "frameworks": {
     "net451": {
       "frameworkAssemblies": {
+        "System.Collections": { "version": "4.0.0.0", "type": "build" },
+        "System.ComponentModel.DataAnnotations": "4.0.0.0",
```
Comment (natemcmaster) [inline src/EntityFramework.Core/project.json]: Shouldn't all the framework assemblies be specified like this one? (this is how we had it in https://github.com/aspnet/EntityFramework/blob/36c7ef55036b02792657563786fba4f0b59e4ebf/src/EntityFramework.Core/project.json#L48)

Comment (natemcmaster) [inline src/EntityFramework.Core/project.json]: Nvm, was looking at the wrong section.

**Comment by natemcmaster** [comment]
Why do we need dnx451?

**Comment by pranavkm** [comment]
@natemcmaster  https://github.com/aspnet/EntityFramework/pull/3504#discussion_r43027248

**Comment by natemcmaster** [comment]
@pranavkm it seems redundant to have `dnx451` and `net451`, but if this is what it takes, then :shipit:. I assume we will cleanup for RC2?

**Comment by davidfowl** [comment]
Must fix for RC2

**Comment by muratg** [comment]
@pranavkm please file a bug in RC2 for the cleanup.

**Comment by pranavkm** [comment]
Wouldn't RC2 have the same issues if we are targeting RTM UAP templates?

**Comment by natemcmaster** [comment]
:crystal_ball:  Hard to know when UAP will update their templates to use generations. When they do, we may be able to drop the "netcore50" TFM altogether.

We definitely will need to cleanup the "dnx451" TFM by RC2.

**Comment by pranavkm** [comment]
Follow up: https://github.com/aspnet/Universe/issues/315

**Comment by natemcmaster** [comment]
Merged with 951933fc0c72e8bcd5795648fa204de7eefcd090.

**Pull request closed without merging**